lib/node_modules/@stdlib/math/special/floor/lib/types.js

```diff
@@ -0,0 +1,88 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/* This is a generated file. Do not edit directly. */
+
+/* eslint-disable array-element-newline */
+
+'use strict';
+
+// MODULES //
+
+var dtypes = require( '@stdlib/ndarray/dtypes' );
+
+
+// MAIN //
+
+var types = [
+	// complex64 (2)
+	dtypes.complex64, dtypes.complex64,
+	dtypes.complex64, dtypes.complex128,
+
+	// complex128 (1)
+	dtypes.complex128, dtypes.complex128,
```

Comment:
We should be able to downcast from `complex128` to `complex64`. Same for other floating-point dtypes. We should support "mostly safe" casts.